runtime/Cpp/runtime/src/NoViableAltException.cpp

```diff
@@ -22,13 +22,13 @@ NoViableAltException::NoViableAltException(Parser *recognizer, TokenStream *inpu
 
 NoViableAltException::~NoViableAltException() {
   if (_deleteConfigs)
-    delete _deadEndConfigs;
+    _deadEndConfigs = nullptr;
```

Comment:
I think the assignment to `nullptr` triggers a `delete` regardless of the `use_count()`. If I just gut the destructor (make it an empty function), then your code works for me.